### PR TITLE
Tensor-based RGBD Odometry

### PR DESCRIPTION
**Pull request by theNded**
This PR brings initial point-to-plane RGBD (in fact depth only) odometry. Initial tests show a reasonable convergence in real-world and synthetic depth images.

TODO:
- [x] cpp example
- [x] better unit tests
- [ ] pybind

Future work:
- better reduction for acceleration

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3085)

### CODE REVIEW

**Review by yxlao — CHANGES_REQUESTED** [review]
Sorry for the delay. LGTM in general after minor changes.

Do you have some rough performance numbers for New CPU v.s. New GPU v.s. Legacy CPU?

> Reviewed 5 of 25 files at r8, 13 of 14 files at r9, 1 of 1 files at r10.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3085#-:-MVabvPsBq6hdElOtWpR:b-njj258)* status: all files reviewed, 11 unresolved discussions (waiting on @reyanshsolis and @theNded)

___
*[cpp/open3d/t/geometry/kernel/PointCloudImpl.h, line 114 at r9](https://reviewable.io/reviews/intel-isl/open3d/3085#-MUhMD1a2lfDwnqWmA47:-MUhMD1a2lfDwnqWmA48:b-r1ne6k) ([raw file](https://github.com/intel-isl/open3d/blob/611958a5b5f3625ca73989223caf04c1a654123f/cpp/open3d/t/geometry/kernel/PointCloudImpl.h#L114)):*
> ```objc
> #endif
>     points = points.Slice(0, 0, total_pts_count);
> }  // namespace pointcloud
> ```

extra?
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryCUDA.cu, line 28 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVabECoDPkkY9EjO8qP:-MVabECoDPkkY9EjO8qQ:b-9tpqlx) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryCUDA.cu#L28)):*
> ```Cuda
> 
> #include "open3d/core/kernel/CUDALauncher.cuh"
> #include "open3d/t/pipelines/kernel/RGBDOdometryImpl.h"
> ```

add empty line
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h, line 186 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVabmSa80PHL4vUC0gg:-MVabmSa80PHL4vUC0gh:bokxkjo) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h#L186)):*
> ```objc
>     residual = core::Tensor::Zeros({}, core::Dtype::Float32, device);
> 
>     float* AtA_local_ptr = static_cast<float*>(AtA.GetDataPtr());
> ```

these can be `GetDataPtr<float>()`
___
*[cpp/open3d/t/pipelines/odometry/RGBDOdometry.h, line 60 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaXPdY0nOGyeotfH0f:-MVaXPdY0nOGyeotfH0g:bqrdg86) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/odometry/RGBDOdometry.h#L60)):*
> ```objc
> 
> /// Create a vertex map (image) from a depth image.
> core::Tensor CreateVertexMap(const t::geometry::Image& depth,
> ```

Typically, do users directly call`CreateNormalMap`, `CreateNormalMap`, ... ?
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 58 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaUr2I9s2f6HbrWAQU:-MVaUr2I9s2f6HbrWAQV:b9h0k0g) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L58)):*
> ```cpp
>     auto focal_length = intrinsic.GetFocalLength();
>     auto principal_point = intrinsic.GetPrincipalPoint();
>     return core::Tensor(
> ```

`core::Tensor::Init<float>({{a, b, c}, {d, e, f}, {g, h, i}});` wiil initialize a 3x3 tensor more conveniently
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 75 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaV_8r12hjUoCo8UD1:-MVaV_8r12hjUoCo8UD2:b-3jg1wp) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L75)):*
> ```cpp
> 
>     auto depth_legacy =
>             io::CreateImageFromFile(std::string(TEST_DATA_DIR) + "/depth.png");
> ```

is `cpp/open3d/t/io/ImageIO.h` now usable?
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 106 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaViNG35v3lYFoe4W_:-MVaViNG35v3lYFoe4Wa:b-nhlzj) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L106)):*
> ```cpp
> 
>     t::geometry::Image depth =
>             t::geometry::Image::FromLegacyImage(*depth_legacy, device);
> ```

same, is `cpp/open3d/t/io/ImageIO.h` now usable?
___
*[examples/cpp/TOdometryRGBD.cpp, line 75 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaU_Yc4eI0qBaoyPos:-MVaU_Yc4eI0qBaoyPot:b9h0k0g) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/cpp/TOdometryRGBD.cpp#L75)):*
> ```cpp
>     auto focal_length = intrinsic.GetFocalLength();
>     auto principal_point = intrinsic.GetPrincipalPoint();
>     core::Tensor intrinsic_t = core::Tensor(
> ```

`core::Tensor::Init<float>({{a, b, c}, {d, e, f}, {g, h, i}});` wiil initialize a 3x3 tensor more conveniently
___
*[examples/cpp/TOdometryRGBD.cpp, line 90 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaV3tj0LU5OblLWgEa:-MVaV3tj0LU5OblLWgEb:bz6jjin) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/cpp/TOdometryRGBD.cpp#L90)):*
> ```cpp
> 
>     // Read input
>     auto src_depth_legacy = io::CreateImageFromFile(src_depth_path);
> ```

`cpp/open3d/t/io/ImageIO.h` is readly, we can switch to that
___
*[examples/test_data/normal_map.npy, line 0 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaSVvF6sZHkj41LEJl:-MVaSVvF6sZHkj41LEJm:biuxxxk) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/test_data/normal_map.npy#L0)):*
Move this to https://github.com/intel-isl/open3d_downloads
___
*[examples/test_data/vertex_map.npy, line 0 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaTWqB1igg-R562rTE:-MVaTWqB1igg-R562rTF:b-4pci5a) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/test_data/vertex_map.npy#L0)):*
Same, move this to https://github.com/intel-isl/open3d_downloads

**Review by reyanshsolis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3085)* status: all files reviewed, 15 unresolved discussions (waiting on @theNded)

___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h, line 255 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVai9HN3ConITatoDVq:-MVai9HN3ConITatoDVr:b-n8x135) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h#L255)):*
> ```objc
>                 atomicAdd(count_ptr, 1);
> #else
> #pragma omp critical
> ```

We may use the tbb vector reduction design (maybe later) to avoid critical construct.
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h, line 276 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaj0Wt-zEVrQ_C9II4:-MVaj0Wt-zEVrQ_C9II5:bu056n) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h#L276)):*
> ```objc
>     // Solve on CPU with double to ensure precision.
>     core::Device host(core::Device("CPU:0"));
>     delta = AtA.To(host, core::Dtype::Float64)
> ```

As we now support Float64 in PoseToTransformation, do we still want to send it to CPU and convert it back to Float64 here?
Also, would using Float64 for AtA be any better?
___
*[cpp/open3d/t/pipelines/odometry/RGBDOdometry.cpp, line 142 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaerht6PeI5VAombuc:-MVaerht6PeI5VAombud:b-hfm0fz) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/odometry/RGBDOdometry.cpp#L142)):*
> ```cpp
> 
>     core::Tensor T_delta_inv = pipelines::kernel::PoseToTransformation(
>             se3_delta.To(core::Dtype::Float32));
> ```

Same. [Float64 in PoseToTransformation], do we still want to convert to Float32 here?
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 75 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVafMs-39xGRJUBsTma:-MVafMs-39xGRJUBsTmb:b-gamua1) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L75)):*
> ```
> (TEST_DATA_DIR) + "/depth.png"
> ```
`io::CreateImageFromFile(TEST_DATA_DIR "/depth.png");` also works, but `(std::string(TEST_DATA_DIR) + "/depth.png)"` might look more intuitive. 
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 217 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVagamx-CuuiFZLanMV:-MVagamx-CuuiFZLanMW:b-hkcfom) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L217)):*
> ```cpp
> 
>     core::Device host("CPU:0");
>     core::Tensor T0(
> ```

`core::Tensor TO = Tensor::Init<double>()` might be more convinient. 
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 226 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVageh714YDTo5Vwd4W:-MVageh714YDTo5Vwd4X:b-7jl271) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L226)):*
> ```cpp
>                                 0.0, 0.0, 1.0},
>             {4, 4}, core::Dtype::Float64, host);
>     core::Tensor T2(
> ```

same.

File: cpp/open3d/t/pipelines/kernel/RGBDOdometry.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+void ComputePosePointToPlane(const core::Tensor &source_vertex_map,
+                             const core::Tensor &target_vertex_map,
+                             const core::Tensor &source_normal_map,
+                             const core::Tensor &intrinsics,
+                             const core::Tensor &init_source_to_target,
+                             core::Tensor &delta,
+                             core::Tensor &residual,
+                             float depth_diff);
+
+void CreateVertexMapCPU(const core::Tensor &depth_map,
```
Comment (reyanshsolis) [inline cpp/open3d/t/pipelines/kernel/RGBDOdometry.h:54]: Could we hide `CPU` and `CUDA` functions from the public-facing header?
I understand in the naive approach of moving the declarations to `Impl.h`, and implementations of `CPU` and `CUDA` to `CPU.cpp` and `CUDA.cu` the re-usability design of code using #ifdef blocks might be compromised.

File: cpp/open3d/t/pipelines/kernel/RGBDOdometry.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#pragma once
+
+#include "open3d/core/Tensor.h"
+
+namespace open3d {
+namespace t {
+namespace pipelines {
+namespace kernel {
+namespace odometry {
+
```
Comment (reyanshsolis) [inline cpp/open3d/t/pipelines/kernel/RGBDOdometry.h:36]: May we add some brief documentation with `\param` description, for `ComputeVertexMap` `ComputeNormalMap` and `ComputePosePointToPlane` ?

**Review by theNded — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3085)* status: 12 of 20 files reviewed, 17 unresolved discussions (waiting on @reyanshsolis and @yxlao)

___
*[cpp/open3d/t/geometry/kernel/PointCloudImpl.h, line 114 at r9](https://reviewable.io/reviews/intel-isl/open3d/3085#-MUhMD1a2lfDwnqWmA47:-MVwfI5MDLSy5_C_eiTO:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/611958a5b5f3625ca73989223caf04c1a654123f/cpp/open3d/t/geometry/kernel/PointCloudImpl.h#L114)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

extra?
</blockquote></details>

Done.
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometry.h, line 36 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MUG8m-B4aSnu-s0P15t-r10-36:-MVwfo6J3xB4qnQwg_6n:b8rsg0k) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometry.h#L36)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

May we add some brief documentation with `\param` description, for `ComputeVertexMap` `ComputeNormalMap` and `ComputePosePointToPlane` ?
</blockquote></details>

We generally don't put comments for kernel functions but for their wrappers.
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometry.h, line 54 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MUG8m-B4aSnu-s0P15t-r10-54:-MVwgX5c3u5wemmhuaHN:b-dndtoj) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometry.h#L54)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

Could we hide `CPU` and `CUDA` functions from the public-facing header?
I understand in the approach of moving the declarations to `Impl.h`, and implementations of `CPU` and `CUDA` to `CPU.cpp` and `CUDA.cu` the re-usability design of code using #ifdef blocks might be compromised. 

</blockquote></details>

I think this is worth discussing. Does it increase or decrease the complexity of the project by hiding some interfaces with the cost of adding one more internal header file. By convention we implemented kernels for tensors in this way, just to keep consistent.
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryCUDA.cu, line 28 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVabECoDPkkY9EjO8qP:-MVwkoCd7euOdEsMtD3H:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryCUDA.cu#L28)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

add empty line
</blockquote></details>

Done.
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h, line 186 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVabmSa80PHL4vUC0gg:-MVwkozeDVDlr30TMJdl:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h#L186)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

these can be `GetDataPtr<float>()`
</blockquote></details>

Done.
___
*[cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h, line 276 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaj0Wt-zEVrQ_C9II4:-MVwgzIH8KpmV6dBib0i:bky6zyb) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/kernel/RGBDOdometryImpl.h#L276)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

As we now support Float64 in PoseToTransformation, do we still want to send it to CPU and convert it back to Float64 here?
Also, would using Float64 for AtA be any better?
</blockquote></details>

CPU version is generally better, and for the small linear system my intuition is that it will be faster on CPU. CUDA solvers might be slow due to the overhead in calling. The general usage of CUDA linear algebra should be large matrix computation or parallel small matrix computations.
___
*[cpp/open3d/t/pipelines/odometry/RGBDOdometry.h, line 60 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaXPdY0nOGyeotfH0f:-MVwhOMY11kmijZ3HYmS:b-ptxp2r) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/odometry/RGBDOdometry.h#L60)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Typically, do users directly call`CreateNormalMap`, `CreateNormalMap`, ... ?
</blockquote></details>

Not really, but it will be very helpful for testing, profiling, and debugging. We can hide the interface once everything is stable.
___
*[cpp/open3d/t/pipelines/odometry/RGBDOdometry.cpp, line 142 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaerht6PeI5VAombuc:-MVxHmcO2w3hxPo96RqO:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/open3d/t/pipelines/odometry/RGBDOdometry.cpp#L142)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

Same. [Float64 in PoseToTransformation], do we still want to convert to Float32 here?
</blockquote></details>

Done.
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 58 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaUr2I9s2f6HbrWAQU:-MVxfRRN6s4uFj0tI_k7:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L58)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

`core::Tensor::Init<float>({{a, b, c}, {d, e, f}, {g, h, i}});` wiil initialize a 3x3 tensor more conveniently
</blockquote></details>

Done.
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 75 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaV_8r12hjUoCo8UD1:-MVxg2LH3nJ4c_8X8NvA:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L75)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

is `cpp/open3d/t/io/ImageIO.h` now usable?
</blockquote></details>

Done.
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 106 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaViNG35v3lYFoe4W_:-MVxibxGD8hCPlpZS6Om:b-wua3vj) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L106)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

same, is `cpp/open3d/t/io/ImageIO.h` now usable?
</blockquote></details>

Upgraded.
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 217 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVagamx-CuuiFZLanMV:-MVxie-4-ZpHYy-d67cA:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L217)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

`core::Tensor TO = Tensor::Init<double>()` might be more convinient. 
</blockquote></details>

Done.
___
*[cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp, line 226 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVageh714YDTo5Vwd4W:-MVxieayEhTFS_Lud7FA:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/cpp/tests/t/pipelines/odometry/RGBDOdometry.cpp#L226)):*
<details><summary><i>Previously, reyanshsolis (Rishabh Singh) wrote…</i></summary><blockquote>

same.
</blockquote></details>

Done.
___
*[examples/cpp/TOdometryRGBD.cpp, line 75 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaU_Yc4eI0qBaoyPos:-MVxifBZ-S6snZ0-6utY:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/cpp/TOdometryRGBD.cpp#L75)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

`core::Tensor::Init<float>({{a, b, c}, {d, e, f}, {g, h, i}});` wiil initialize a 3x3 tensor more conveniently
</blockquote></details>

Done.
___
*[examples/cpp/TOdometryRGBD.cpp, line 90 at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaV3tj0LU5OblLWgEa:-MVxif_Q3km8npMRz4dy:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/cpp/TOdometryRGBD.cpp#L90)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

`cpp/open3d/t/io/ImageIO.h` is readly, we can switch to that
</blockquote></details>

Done.
___
*[examples/test_data/normal_map.npy, line  at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaSVvF6sZHkj41LEJl:-MVxmU0E9E71TSdyb71f:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/test_data/normal_map.npy#L)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Move this to https://github.com/intel-isl/open3d_downloads
</blockquote></details>

Done.
___
*[examples/test_data/vertex_map.npy, line  at r10](https://reviewable.io/reviews/intel-isl/open3d/3085#-MVaTWqB1igg-R562rTE:-MVxmTpC0_bMwgyTSjXR:b-896fix) ([raw file](https://github.com/intel-isl/open3d/blob/ad97d22622fe3e5155eac26c8050113ac8980d7e/examples/test_data/vertex_map.npy#L)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

Same, move this to https://github.com/intel-isl/open3d_downloads
</blockquote></details>

Done.

**Review by yxlao — APPROVED** [review]
> Reviewed 11 of 11 files at r11.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3085)* status: all files reviewed, 10 unresolved discussions (waiting on @reyanshsolis and @yxlao)

**Pull request merged**